apps/aecore/src/aec_spend_tx.erl

```diff
@@ -49,21 +49,18 @@ new(#{sender := SenderPubkey,
       recipient := RecipientPubkey,
       amount := Amount,
       fee := Fee,
-      ttl := TTL,
       nonce := Nonce,
-      payload := Payload}) when is_integer(Amount), Amount >= 0,
-                                is_integer(Nonce), Nonce >= 0,
-                                is_integer(Fee), Fee >= 0,
-                                is_integer(TTL), TTL >= 0,
-                                is_binary(SenderPubkey),
-                                is_binary(RecipientPubkey),
-                                is_binary(Payload)
-                                ->
+      payload := Payload} = Args) when is_integer(Amount), Amount >= 0,
+                                       is_integer(Nonce), Nonce >= 0,
+                                       is_integer(Fee), Fee >= 0,
+                                       is_binary(SenderPubkey),
+                                       is_binary(RecipientPubkey),
+                                       is_binary(Payload) ->
     Tx = #spend_tx{sender = SenderPubkey,
                    recipient = RecipientPubkey,
                    amount = Amount,
                    fee = Fee,
-                   ttl = TTL,
+                   ttl = maps:get(ttl, Args, 0),
```

Comment:
This is removing the check that value must be integer. Though :new shall be called on already sanitized data so shall be ok.